dotNES/UI.cs

```diff
@@ -253,12 +275,31 @@ private void UI_MouseClick(object sender, MouseEventArgs e)
                                 y.Click += delegate { activeSpeed = speed; };
                             }));
                     }),
-                    new Item("&Reset..."),
+                    new Item("&Reset", x => x.Click += Reset)
                 }
             };
             cm.Show(this, new Point(e.X, e.Y));
         }
 
+        private void LoadRom(object sender, EventArgs e)
+        {
+            var ofd = new OpenFileDialog
+            {
+                Filter = "rom|*.nes"
+            };
+            if (DialogResult.OK == ofd.ShowDialog())
+            {
+                BootCartridge(ofd.FileName);
+                AllowDrop = false;
+            }
+        }
+
+        private void Reset(object sender, EventArgs e)
+        {
+            Process.Start(Application.ExecutablePath);
+            Close();
```

Comment:
I don't think this has the intended effect, since it actually causes the current cartridge's state to be saved rather than reset (https://github.com/Xyene/Emulator.NES/blob/master/dotNES/UI.cs#L175). I think stopping the `renderingThread`, and setting `gameStarted` to false should achieve the intended effect without restarting the application.